golem/managers/proposal/plugins/scoring/scoring_buffer.py

```diff
@@ -55,33 +75,34 @@ async def stop(self) -> None:
         await super().stop()
 
         if self._background_loop_task is not None:
-            await cancel_and_await(self._background_loop_task)
+            await ensure_cancelled(self._background_loop_task)
             self._background_loop_task = None
 
-    async def _on_item_added(self, item: Proposal):
+    async def _on_added(self, proposal: Proposal) -> None:
         pass  # explicit no-op
 
     async def _background_loop(self) -> None:
         while True:
-            logger.debug("Waiting for any items to score...")
-            await self._buffer.wait_for_any_items()
-            logger.debug("Waiting for any items to score done, items are available for scoring")
-
-            logger.debug("Waiting for more items up to %s...", self._update_interval)
-            items = await self._buffer.get_all_requested(self._update_interval)
-            logger.debug("Waiting for more items done, %d new items will be scored", len(items))
+            logger.debug(
+                "Waiting for any proposals to score with debounce of `%s`...",
```

Comment:
I'm unsure if "debounce" is the most obvious word in this context... I'm not even sure what it means here... is this the minimum number of proposals to wait for until any scoring will happen?

Reply:
`Debounce` is fairly settled term, it always refers to time. Minimum amount of time to catch more proposals before scoring will happen.

Reply:
far from perfectly clear but indeed, I hadn't looked at the type there